bundles/org.openhab.binding.anthem/src/main/java/org/openhab/binding/anthem/internal/handler/AnthemConnectionManager.java

```diff
@@ -0,0 +1,433 @@
+        connect();
+    }
+
+    private void senderThreadJob() {
+        logger.debug("Sender thread started");
+        try {
+            while (!Thread.currentThread().isInterrupted() && writer != null) {
+                AnthemCommand command = sendQueue.take();
+                logger.debug("Sender thread writing command: {}", command);
+                try {
+                    BufferedWriter localWriter = this.writer;
+                    if (localWriter != null) {
+                        localWriter.write(command.toString());
+                        localWriter.flush();
+                    }
+                } catch (InterruptedIOException e) {
+                    logger.debug("Interrupted while sending command");
+                    handler.updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Interrupted");
+                    break;
+                } catch (IOException e) {
+                    logger.warn("Communication error, will try to reconnect. Error: {}", e.getMessage());
+                    handler.updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                    // Requeue the command and try to reconnect
+                    sendQueue.add(command);
+                    reconnect();
+                    break;
+                }
+                // Introduce delay to throttle the send rate
+                if (commandDelayMsec > 0) {
+                    Thread.sleep(commandDelayMsec);
+                }
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } finally {
+            logger.debug("Sender thread exiting");
+        }
+    }
+
+    private void readerThreadJob() {
+        logger.debug("Reader thread started");
+        StringBuffer sbReader = new StringBuffer();
+        try {
+            char c;
+            String command;
+            BufferedReader localReader = this.reader;
+            while (!Thread.interrupted() && localReader != null) {
+                c = (char) localReader.read();
+                sbReader.append(c);
+                if (c == COMMAND_TERMINATION_CHAR) {
+                    command = sbReader.toString();
+                    logger.debug("Reader thread sending command to parser: {}", command);
+                    messageParser.parseMessage(command);
+                    sbReader.setLength(0);
+                }
+            }
+        } catch (InterruptedIOException e) {
+            logger.debug("Interrupted while reading");
+            handler.updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Interrupted");
+        } catch (IOException e) {
+            logger.debug("I/O error while reading from socket: {}", e.getMessage());
+            handler.updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "I/O Error");
+        } catch (RuntimeException e) {
```

Comment:
I was worried about an unexpected issue when parsing a command returned from the device, and thought it might be useful to see the stack trace. WDYT?

Reply:
Wouldn't this be a bug in the command parser class then? It doesn't seem to me than anything external is involved and can throw runtime exceptions here? I just quickly went through that class and you are already catching many exceptions (like array out of bounds) here, so seems quite well guarded. Perhaps consider some unit test coverage for corner cases.

Reply:
Yes, it definitely would be a bug in the parser if the AV processor sent something that would cause a runtime exception in the parser. I also was pretty confident that the parser was catching most of the expected exceptions. I was just worried about something quite unexpected. I'll remove the RuntimeException in the next iteration.

Reply:
RuntimeException catch block removed